### PR TITLE
4.0.0.pre10!

### PR DESCRIPTION
**Pull request by jasonrclark**
Handy checklists are handy!

## For the Release

* [x] Run through the samples on all platforms
  * [x] Mac
  * [x] Windows (actually running samples needs #1443, but they worked fine)
  * [x] Linux
* [x] Run packaging on all platforms from installed `shoes` command. Ensure jars and apps run on all applicable platforms
  * [x] Mac
  * [x] Windows (#1448)
  * [x] Linux
* [x] Increment version number (edit VERSION and run `rake update_version` to update numbers)
* [x] Write release blog post (https://github.com/shoes/shoesrb.com/pull/82)
* [x] Merge this PR
* [ ] Release `furoshiki` 0.5.0 (merging https://github.com/shoes/furoshiki/pull/32 fixes broken build here)
* [ ] Release the shoooooooes! (`rake release:all`)
* [ ] Send message to mailing list
* [ ] Release blog post
* [ ] :tada: :bell: :triumph:

## For the _Next_ Release
* [ ] Once finished celebrating, look for gem updates!

### CODE REVIEW

**Comment by PragTob** [comment]
Great stuff, thank you Jason!

File: CHANGELOG [hunk 1]
```diff
@@ -1,3 +1,28 @@
+v4.0.0.pre10 (40 commits) (Jun 3, 2017)
+--------------------------------------------
+
+Improvements (_)
+----------------
+
+* Support packaging Linux app [d22b561]
+* Support packaging Windows apps [d22b561]
```
Comment (PragTob) [inline CHANGELOG]: it doesn't look like MANY items, but these are HUGGGGEEEE!

**Comment by jasonrclark** [comment]
Ugggggggh. So went about my merry way trying to test things, and all packaging of every sort was busted on Windows 😭 

I don't have time tonight to get this winnowed down to a set of repro steps I'm happy with posting elsewhere (still too many big steps), but there's a change between JRuby 9.1.8.0 and 9.1.10.0. Under 9.1.10.0, our require for `shoes` when initializing in the package fails to find the file on Windows where before it did. The `Gem.paths.path` it's looking at is identical between versions, so either JRuby or maybe [an updated version of Rubygems](https://github.com/jruby/jruby/pull/4537) isn't behaving the same.

FWIW, it appears that if I pinned the release to use `jruby-jars-9.1.8.0` we'd be back to working (even if you built the package with a newer version of JRuby). This bears more testing, but might give a shorter path to getting `pre10` locked down while I run down the other problems.

**Comment by jasonrclark** [comment]
Ok last bit for night... wonder if related to https://github.com/jruby/jruby/pull/4543

Next steps will try to repri with plain warbler + JRuby versions

**Comment by jasonrclark** [comment]
All right, managed to get that pinned down more closely (just bare warbler + varying JRuby version) enough to [file an issue with JRuby](https://github.com/jruby/jruby/issues/4).

Done for the night, but next up I'll PR pinning to 9.1.8.0 for the embedded `jruby-jars` that we pull along with `shoes-package` and crew, and see if I can coerce that to just package up the working version and disregard your running version of JRuby (which has proven workable in my repro case testing). If so, I can get this unblocked without waiting on the other issue to resolve.

**Comment by PragTob** [comment]
I think jason meant to reference jruby/jruby#4653 which should be resolved in the newest JRuby, should get some of those sweet JRuby updates going :)

**Comment by jasonrclark** [comment]
JRuby 9.1.11.0 did indeed address it (and thanks for the typo catch @PragTob!)

I've updated it on this branch and finished testing. Found one issue with apps for Mac/Linux packaged on Windows, but I'm going to address it separately.

Writing things up, will give them another read-over tomorrow and hit the button to get this thing out the door!

**Comment by PragTob** [comment]
Bumped to newer JRuby (there was some quickfix release for 12)

**Comment by jasonrclark** [comment]
Thanks for the 🏃 💨 work on that @PragTob! All looks good so I'm going for it.

**Comment by PragTob** [comment]
:tada:

**Pull request merged**